.github/workflows/code-freeze.yml

```diff
@@ -0,0 +1,84 @@
+name: Code Freeze CI
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+  merge_group:
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  RELEASE_LABEL_NAME: release
+  ON_FREEZE_MODE: false
+
+jobs:
+  code_freeze_pr:
+    name: Code Freeze Pull Request
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout To Repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
+        with:
+          fetch-depth: 0
+
+      - name: Block Rust Changes On Code Freeze
+        if: ${{ env.ON_FREEZE_MODE == 'true' }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -ex
+
+          ockam_team_members="""
+            etorreborre
+            mrinalwadhwa
+            davide-baldo
+            hairyhum
+            glenngillen
+            CAOakleyII
+            BeenzSyed
+            SanjoDeundiak
+            adrianbenavides
+            polvorin
+            mattgreg
+            metaclips
+            mszpakowski
```

Comment:
This is hackish, we currently can't get team members using github api with GITHUB_TOKEN so we have to hardcode members more here https://github.com/tspascoal/get-user-teams-membership#requirements